tendenci/apps/directories/views.py

```diff
@@ -314,6 +314,10 @@ def edit(request, id, form_class=DirectoryForm, template_name="directories/edit.
                 except IOError:
                     pass
                     #directory.logo = None
+            
+            # set the expiration date in case activation date and/or requested_duration have changed
+            directory.expiration_dt = directory.activation_dt + timedelta(days=directory.requested_duration)
```

Comment:
Please add an if statement to make sure "directory.activation_dt" is not None. Also, this should only be available to superuser as regular users will not be able to edit the requested duration.

Reply:
Also, please check if "directory.pricing" exists or not.

Reply:
@jennyq I've put in checks for directory.activation_dt AND for directory.requested_duration.

I had a look at the directory.add method and there are way more business rules in that around pricing, require_payment etc than there are in the edit method.  It occurs to me that maybe those rules should be pulled out into a single "gatekeeper" method that is called by both the add() and edit() methods?